src/Emulator/Peripherals/Peripherals/Timers/STM32F4_Watchdog.cs

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) 2021 Zisis Adamos
+//
+//  This file is licensed under the MIT License.
+//  Full license text is available in 'licenses/MIT.txt'.
+//
+
+using System;
+using Antmicro.Renode.Core;
+using Antmicro.Renode.Core.Structure.Registers;
+using Antmicro.Renode.Time;
+using Antmicro.Renode.Logging;
+
+namespace Antmicro.Renode.Peripherals.Timers
+{
+    public class STM32F4_Watchdog : BasicDoubleWordPeripheral, IKnownSize
+    {
+        //TODO: Could propably get osc_frequency from RCC.
+        //TODO: Stop timer on debug stop.
+        //TODO: Use RCC to set restart cause.
+        public STM32F4_Watchdog(Machine machine, long osc_frequency) : base(machine)
+        {
+            watchdogTimer = new LimitTimer(machine.ClockSource, osc_frequency, this, "STM32_IWDG", watchdogDefaultRLRValue, workMode: WorkMode.OneShot, enabled: false, eventEnabled: false);
```

Comment:
I think you can enable generating of events right away in the constructor. It won't fire anyway before enabling the timer itself.